libraries/nouveau-handler/bnd.bnd

```diff
@@ -0,0 +1,26 @@
+#
```

Comment:
I think we shouldn't use **bnd** file here because we have already deleted them in other places

Reply:
Correct. We did goodbye for OSGI bundles ( and i expect not see it anymore )

Reply:
I expected the get a merge conflict at rebase. I will remove them and update the branch. Thank you!

Reply:
Removed.

Reply:
@GMishx Now it looks good to me. Just a small question, do I need any configuration to make it work?

Reply:
Thanks @hoangnt2

You do not need to configure anything differently in SW360, just need to install couchdb with Nouveau.

You can do so with either of following methods (will recommend the helper script which has minimal manual steps):
1. Use the helper install script from branch: https://github.com/eclipse-sw360/sw360/blob/0a0ce15327d902b2aa119ac32955c6434950dcae/scripts/install-couchdb-nouveau.sh
2. Use the Dockerfile from branch to build a couchdb image with nouveau: https://github.com/eclipse-sw360/sw360/blob/0a0ce15327d902b2aa119ac32955c6434950dcae/scripts/couchdb/Dockerfile
3. Download the source of couchdb and configure it using `./configure --enable-nouveau` flag and then install it as usual.
    - Sometimes the file `./rel/couchdb/etc/nouveau.yaml` has to be manually placed in `/opt/couchdb/etc/nouveau.yaml`
    - And the path of `rootDir` in the yaml file set to `./data/nouveau` manually.
    - And following additional service has to be created to start the nouveau server at `/lib/systemd/system/nouveau.service` and enabled with `sudo systemctl daemon-reload` `sudo systemctl enable --now nouveau.service`
    ```
    [Unit]
    Description=Apache CouchDB Nouveau
    Wants=couchdb.service
    After=couchdb.service
    
    [Service]
    #EnvironmentFile=-/etc/default/couchdb
    #RuntimeDirectory=/opt/couchdb
    WorkingDirectory=/opt/couchdb
    User=couchdb
    Group=couchdb
    ExecStart=/opt/couchdb/nouveau/bin/nouveau server /opt/couchdb/etc/nouveau.yaml
    Restart=always
    LimitNOFILE=65536
    
    [Install]
    WantedBy=multi-user.target
    ```

Reply:
@GMishx, Thanks a lot, I'll give it a try

Reply:
@GMishx, 
I tried to build couchdb image using your Docker file. Then I got an error (--enable-nouveau option no longer exists).
I looked at your Docker file. You have cloned the couchdb source code from the main branch of [https://github.com/apache/couchdb.git](https://github.com/apache/couchdb.git).
But currently, the main branch has a lot of changes.
I also tried checkout from main branch to commit to version 3.4.0 (26a718e98cd3500a487def1e89407f4f368feae2), but I got another error.
Here is my log file. Please help me to check it.
[couchdb-build.log](https://github.com/user-attachments/files/16609854/couchdb-build.log)

Reply:
Ah, from the logs looks like fauxton compilation failed due to some syntax error. I will try to build it on the latest release [3.3.3.post4](https://github.com/apache/couchdb/releases/tag/3.3.3.post4) once as there are changes done in nouveau after 3.3.3 release which are required for SW360.

Reply:
Hi @hoangnt2 , I managed to get the Docker build working again. The option `--enable-nouveau` has been replaced with `--with-nouveau` and the dependencies of the CouchDB has been changed since. I have installed the required version of NPM and Erlang/OTP and have tagged the repositories to current commit IDs to prevent this issue in future.

Can you please try to build the image again?

Reply:
Would be solved by #2627 thus marking this conversation as resolved.